include/nbl/asset/utils/CGLSLCompiler.h

```diff
@@ -0,0 +1,174 @@
+
+		CGLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system);
+
+		struct SOptions : IShaderCompiler::SOptions
+		{
+			void setCommonData(const IShaderCompiler::SOptions& opt)
+			{
+				static_cast<IShaderCompiler::SOptions&>(*this) = opt;
+			}
+
+			virtual IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_GLSL; };
+		};
+
+		/**
+		If options.stage is ESS_UNKNOWN, then compiler will try to deduce shader stage from #pragma annotation, i.e.:
+		#pragma shader_stage(vertex),       or
+		#pragma shader_stage(tesscontrol),  or
+		#pragma shader_stage(tesseval),     or
+		#pragma shader_stage(geometry),     or
+		#pragma shader_stage(fragment),     or
+		#pragma shader_stage(compute)
+
+		Such annotation should be placed right after #version directive.
+
+		This function does NOT process #include directives! Use resolveIncludeDirectives() first.
+
+		@params code high level code
+		@params options see IShaderCompiler::SOptions
+			- entryPoint Must be "main" since shaderc does not allow other entry points for GLSL. Kept with hope that shaderc will drop that requirement.
+
+		@returns Shader containing SPIR-V bytecode.
+		*/
+		core::smart_refctd_ptr<ICPUBuffer> compileToSPIRV(const char* code, const CGLSLCompiler::SOptions& options) const;
+
+		core::smart_refctd_ptr<ICPUShader> createSPIRVShader(const char* code, const CGLSLCompiler::SOptions& options) const;
+
+		core::smart_refctd_ptr<ICPUShader> createSPIRVShader(system::IFile* sourceFile, const CGLSLCompiler::SOptions& options) const;
+
+		// TODO: REMOVE after GL Deprecation
+		core::smart_refctd_ptr<ICPUBuffer> compileSPIRVFromGLSL(
+			const char* _glslCode,
+			IShader::E_SHADER_STAGE _stage,
+			const char* _entryPoint,
+			const char* _compilationId,
+			bool _genDebugInfo = true,
+			std::string* _outAssembly = nullptr,
+			system::logger_opt_ptr logger = nullptr,
+			const E_SPIRV_VERSION targetSpirvVersion = E_SPIRV_VERSION::ESV_1_6) const;
+
+		// TODO: REMOVE after GL Deprecation
+		core::smart_refctd_ptr<ICPUShader> createSPIRVFromGLSL(
+			const char* _glslCode,
+			IShader::E_SHADER_STAGE _stage,
+			const char* _entryPoint,
+			const char* _compilationId,
+			const ISPIRVOptimizer* _opt = nullptr,
+			bool _genDebugInfo = true,
+			std::string* _outAssembly = nullptr,
+			system::logger_opt_ptr logger = nullptr,
+			const E_SPIRV_VERSION targetSpirvVersion = E_SPIRV_VERSION::ESV_1_6) const;
+
+		// TODO: REMOVE after GL Deprecation
+		core::smart_refctd_ptr<ICPUShader> createSPIRVFromGLSL(
+			system::IFile* _sourcefile,
+			IShader::E_SHADER_STAGE _stage,
+			const char* _entryPoint,
+			const char* _compilationId,
+			const ISPIRVOptimizer* _opt = nullptr,
+			bool _genDebugInfo = true,
+			std::string* _outAssembly = nullptr,
+			system::logger_opt_ptr logger = nullptr,
+			const E_SPIRV_VERSION targetSpirvVersion = E_SPIRV_VERSION::ESV_1_6) const;
```

Comment:
nuke them while you're at it, only used by OpenGL backend?

Reply:
no it was used by both backends but I didn't want to change/fix opengl 
I will most probably remove this and comment the places used to ignore compile errors